File: 21551-naveen-siva-kumar/week1/1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int braces_count =0;
        int ans =0;
        //cout<<s.size()-1<<endl;
        for(int i=s.size()-1;i>-1;i--){
            
            if(s[i]==')') {
                braces_count++;
            }
            else if(s[i]=='('){
                braces_count--;
            }
            ans = std::max(ans,braces_count);
        }
        return ans;
    }
};

File: 21551-naveen-siva-kumar/week1/1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        
        Stack braces = new Stack();
        
        char[] str = s.toCharArray();
        
        int ans = 0,count=0;
        
        for(int i=0;i<s.length();i++){
            
            if(str[i]=='('){
                
                braces.push(str[i]);
                count+=1;
            }
            else if(str[i]==')'){
                
                //if(braces.empty()){ continue;}
                ans = Math.max(ans,count);
                braces.pop();
                count--;
            }
            else{
                continue;
            }
        }
        return ans;
        
    }
}

